### PR TITLE
Add support for PHP 8.1

### PR DESCRIPTION
**Pull request by arueckauer**
|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

This PR

- adds PHP 8.1 support
- drops PHP 7.3 support
- upgrades `laminas/laminas-container-config-test` to `0.4`

### CODE REVIEW

**Comment by arueckauer** [comment]
Tests fail on PHP 8.1 lowest due to `Pimple\Container` throwing deprecation notices for implemented `ArrayAccess`. This requires a fix in `pimple/pimple`.

**Review by Ocramius — APPROVED** [review]
LGTM :+1:

**Comment by Ocramius** [comment]
Thanks, @arueckauer!

**Review by Ocramius — CHANGES_REQUESTED** [review]
Ah, never mind, we still have a crash:

```

PHP Fatal error:  During inheritance of ArrayAccess: Uncaught Return type of Pimple\Container::offsetExists($id) should either be compatible with ArrayAccess::offsetExists(mixed $offset): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice
``` 

:|

**Comment by froschdesign** [comment]
@arueckauer 
Pimple must be updated: 

> 3.5.0 (2021-10-28)
> 
> * Add support for PHP 8.1
> * Add support for version 2.0 of PSR-11

https://github.com/silexphp/Pimple/blob/69bba50d1600342deae9dca3372a2b5f1ac32ae6/CHANGELOG#L1-L4

**Comment by froschdesign** [comment]
@arueckauer 
Thanks for your quick response! 👍 

Can you also check the conflicts for `composer.json` and `composer.lock`?

**Comment by arueckauer** [comment]
@froschdesign Forgot to pull upstream. It is solved now (hopefully without messing up history).

File: composer.json [hunk 1]
```diff
@@ -23,8 +23,9 @@
     "extra": {
     },
     "require": {
-        "php": "^7.3 || ~8.0.0",
-        "pimple/pimple": "^3.2.2"
+        "php": "^7.3 || ~8.0.0 || ~8.1.0",
+        "laminas/laminas-zendframework-bridge": "^1.0",
```
Comment (Ocramius) [inline composer.json]: Wait, what happened here? :| 

We should get rid of this, not add it...

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (arueckauer) [reply composer.json]: Upgraded `laminas-container-config-test` to version 0.4 and dropped `laminas/laminas-zendframework-bridge`.

**Comment by arueckauer** [comment]
Looking into providing PHP 8.1 support for `laminas-container-config-test`.

**Comment by arueckauer** [comment]
This PR relies on a PHP 8.1 compatible release of `laminas-container-config-test`, see laminas/laminas-container-config-test#12

**Review by froschdesign — APPROVED** [review]

**Comment by froschdesign** [comment]
@arueckauer 
Thank you for your time and this contribution! 👍

**Pull request merged**